pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -548,6 +548,12 @@ def _save_top_k_checkpoints(self, trainer, pl_module, metrics):
         epoch = metrics.get("epoch")
         step = metrics.get("step")
 
+        # when `val_loss` is being logged and no ModelCheckpoint is being provided
+        # `val_loss` or `checkpoint_on` will be selected for monitor and need to be reduced to
+        # prevent processes divergence
+        if self.monitor in ("val_loss", "checkpoint_on"):
```

Comment:
Shouldn't we reduce it always?

Reply:
Metrics already perform reduction during compute. I think the issue only happens with legacy metrics.

Reply:
I'm not sure about this. I thought the metrics that the checkpoint gets from the trainer/logger connector are already reduced? We shouldn't let the checkpoint have the responsibility to reduce metrics or to assume how. the mean is not always correct.

Reply:
can we get a test case of this legacy metric? It will help debug and figure out why this reduction is necessary